### PR TITLE
Update docs to API version 0.7

### PR DESCRIPTION
**Pull request by AidanDelaney**
Use `pack sbom download` and API 0.7 to demonstrate a CycloneDX sbom
in a buildpack.

### CODE REVIEW

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 1]
```diff
@@ -38,18 +38,77 @@ Processes:
...
 # Append a Bill-of-Materials containing metadata about the provided ruby version
-cat >> "$layersdir/launch.toml" << EOL
-[[bom]]
-name = "ruby"
-[bom.metadata]
-version = "$ruby_version"
+cat >> "$layersdir/ruby.sbom.cdx.json" << EOL
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:70]:
```suggestion
  "specVersion": "1.4",
```

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 2]
```diff
@@ -146,28 +224,41 @@ pack build test-ruby-app --path ./ruby-sample-app --buildpack ./ruby-buildpack
 ```
 <!--+- "{{execute}}"+-->
 
-You should then be able to inspect your Ruby app for its Bill-of-Materials via:
+Viewing your bill-of-materials requires first pushing the built image to a container registry and then querying the bill-of-materials.  If you are a `docker` user, we can experiment with an OCI container registry by running an instance of Docker's `registry:2` (as an aside, the standard Docker daemon is a container registry, but is not fully [OCI compliant](https://opencontainers.org/)).
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:227]: You should be able to view the sbom from the daemon as well if you passed the --local flag.

**Review by samj1912 — COMMENTED** [review]
Looks really good! Some minor changes

**Review by dgsdoug — APPROVED** [review]

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md (hunk 2 above)
Comment (AidanDelaney) [reply content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md]: Thanks. I've updated this and removed all the `registry:2` complexity.

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 3]
```diff
@@ -38,18 +38,77 @@ Processes:
...
+```
+
+We can also add an BOM entry for each dependency listed in `Gemfile.lock`.  Here we use `jq` to add a new record to the `components` array in `bundler.sbom.cdx.json`:
+
+```bash
+crubybom="${layersdir}/ruby.sbom.cdx.json"
+cat >> ${rubybom} << EOL
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:90]:
```suggestion
  "specVersion": "1.4",
```

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 4]
```diff
@@ -129,13 +188,32 @@ EOL
...
-cat >> "$layersdir/launch.toml" << EOL
-[[bom]]
-name = "ruby"
-[bom.metadata]
-version = "$ruby_version"
+rubybom="${layersdir}/ruby.sbom.cdx.json"
+cat >> ${rubybom} << EOL
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:195]:
```suggestion
  "specVersion": "1.4",
```

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 5]
```diff
@@ -146,28 +224,38 @@ pack build test-ruby-app --path ./ruby-sample-app --buildpack ./ruby-buildpack
...
+```
+
 You should find that the included `ruby` version is `2.5.0` as expected.
 
-<!-- test:assert=contains -->
+<!-- test:assert=contains;ignore-lines=... -->
 ```text
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md:248]:
```suggestion
  "specVersion": "1.4",
```

File: katacoda/scenarios/buildpack-author-guide/adding-bill-of-materials.md [hunk 6]
```diff
@@ -34,18 +34,77 @@ Processes:
...
+```
+
+We can also add an BOM entry for each dependency listed in `Gemfile.lock`.  Here we use `jq` to add a new record to the `components` array in `bundler.sbom.cdx.json`:
+
+```bash
+crubybom="${layersdir}/ruby.sbom.cdx.json"
+cat >> ${rubybom} << EOL
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline katacoda/scenarios/buildpack-author-guide/adding-bill-of-materials.md:86]:
```suggestion
  "specVersion": "1.4",
```

File: katacoda/scenarios/buildpack-author-guide/adding-bill-of-materials.md [hunk 7]
```diff
@@ -125,13 +184,32 @@ EOL
...
-cat >> "$layersdir/launch.toml" << EOL
-[[bom]]
-name = "ruby"
-[bom.metadata]
-version = "$ruby_version"
+rubybom="${layersdir}/ruby.sbom.cdx.json"
+cat >> ${rubybom} << EOL
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
```
Comment (samj1912) [inline katacoda/scenarios/buildpack-author-guide/adding-bill-of-materials.md:191]:
```suggestion
  "specVersion": "1.4",
```

File: content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md [hunk 8]
```diff
@@ -38,18 +38,77 @@ Processes:
...
 
-To add the `ruby` version to the output of `pack inspect-image`, we will have to provide a `Bill-of-Materials` (`BOM`) containing this information. You'll need to update the `launch.toml` created at the end of your `build` script:
+To add the `ruby` version to the output of `pack download sbom`, we will have to provide a `Bill-of-Materials` (`BOM`) containing this information. There are three "standard" ways to report SBOM data.  You'll need to choose to use on of CycloneDX, SPDX or Syft update the `ruby.sbom.<ext>` (where `<ext>` is the extension appropriate for your BOM standard, one of `cdx.json`, `spdx.json` or `syft.json`) at the end of your `build` script.  Discussion of which BOM format to choose is outside the scope of this tutorial, but we will note that the SBOM format you choose to use is likely to be the output format of any BOM scanner (eg: [`syft cli`](https://github.com/anchore/syft)) you might choose to use.  In this example we will use the CycloneDX json format.
```
[the remark this answers is not in the document]
Comment (AidanDelaney) [reply content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md]: I've linked out to each, and I've also consistently used SBOM as opposed to BOM.

**Review by samj1912 — APPROVED** [review]

**Pull request merged**